### PR TITLE
picasso

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**

### CODE REVIEW

**Review by pacmanifold — CHANGES_REQUESTED** [review]
Hey @dzmitry-lahoda ! Thanks a lot for your PR. I am not familiar with Picasso only had a quick look [here](https://docs.composable.finance/parachains/picasso-parachain-overview/). I hope that is the correct picasso. I have a few general questions

- Is there any way of integration testing the implementation against a Picasso instance? Either running locally in a docker container or preferably something like osmosis [test-tube](https://github.com/osmosis-labs/test-tube)?
- Does this implement interaction with the [PabloDEX](https://docs.composable.finance/products/pablo-overview)?
- Will the contract address and messaging change in the future? In that case maybe we should consider maintaining the messages as well as the const address in a separate crate

File: cw-dex/src/implementations/picasso/pool.rs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            .collect();
+
+        let msg = super::msg::ExecuteMsg::AddLiquidity {
+            pool_id: self.pool_id,
+            assets: assets,
+            min_mint_amount: min_out,
+            keep_alive: true,
+        };
+        let msg = WasmMsg::Execute {
+            contract_addr: "5w3oyasYQg6vkzwETMqUfvtVM99GQ4Xy8mMdKXMgJZDoRYwg".to_owned(),
```
[the remark this answers is not in the document]
Comment (dzmitry-lahoda) [reply cw-dex/src/implementations/picasso/pool.rs:53]: sure

**Comment by dzmitry-lahoda** [comment]
> Hey @dzmitry-lahoda ! Thanks a lot for your PR. I am not familiar with Picasso only had a quick look [here](https://docs.composable.finance/parachains/picasso-parachain-overview/). I hope that is the correct picasso. I have a few general questions
> 
> * Is there any way of integration testing the implementation against a Picasso instance? Either running locally in a docker container or preferably something like osmosis [test-tube](https://github.com/osmosis-labs/test-tube)?
> * Does this implement interaction with the [PabloDEX](https://docs.composable.finance/products/pablo-overview)?
> * Will the contract address and messaging change in the future? In that case maybe we should consider maintaining the messages as well as the const address in a separate crate
> 
> Also, is the source of the DEX somewhere?

yes. that dex.
we are dotsama, not cosmos, so cosmos tooling does not work with us.
testing like this https://github.com/ComposableFi/composable/blob/main/code/parachain/frame/cosmwasm/cli/README.md

i may end up testing if will use cw dex for our cross chain swaps with osmosis and others.

until cw 2 with no std support, any crate we produce are not compatible in compile time with ci of cosmos people. in september 2023 if cw 2 will merge no std we will produce crate.

source of dex https://github.com/ComposableFi/composable/blob/main/code/parachain/pablo

**Comment by dzmitry-lahoda** [comment]
unrelated deny fails.
linting to fix.

**Comment by pacmanifold** [comment]
hey @dzmitry-lahoda ! sorry for a slow response. very busy this week with some final stuff before one week vacation. I'll be back around 2nd of Aug. I really need to read up a bit more on Dotsama. Would love to take a quick call and discuss how you plan on using the cw-dex implementation and what would be needed from our side for our contracts to support Composable.

**Pull request closed without merging**